python/rgbmatrix/core.pyx

```diff
@@ -83,18 +83,100 @@ cdef class FrameCanvas(Canvas):
         def __get__(self): return (<cppinc.FrameCanvas*>self.__getCanvas()).pwmbits()
         def __set__(self, pwmBits): (<cppinc.FrameCanvas*>self.__getCanvas()).SetPWMBits(pwmBits)
 
+cdef class RGBMatrixOptions:
+    def __cinit__(self):
```

Comment:
The Python way is what the code does now. Adding the arguments definitions to the arg parser and calling the parse function. Sub modules can be given the opportunity to add their own arguments before the actual parsing and then have access to the args instance to retrieve their values.

it's probably not hard to call the C++ parsing (haven't tried). It's that there are few entities that need to deal with this args, set their switch names and display their help.

I don't think RGBMatrixOptions should know or care about args parsing at all, it's a **library class** and args parsing the **application** logic (and it won't solve the original issue anyway). 

I will try to play with it and see if this can be solved between the SampleBase class and the C++ parsing code and that they can play together nicely. But as I said, it's not a regression from the original code and can be treated separately.